js/private/node-patches/src/fs.ts

```diff
@@ -782,26 +833,40 @@ export const escapeFunction = (_roots: string[]) => {
             linkTarget = path.resolve(linkTarget)
         }
 
-        let escapedRoot = undefined
         for (const root of roots) {
             // If the link is in the root check if the realPath has escaped
-            if (isSubPath(root, linkPath) || linkPath == root) {
-                if (!isSubPath(root, linkTarget) && linkTarget != root) {
-                    if (!escapedRoot || escapedRoot.length < root.length) {
-                        // if escaping multiple roots then choose the longest one
-                        escapedRoot = root
-                    }
-                }
+            if (isSubPath(root, linkPath) && !isSubPath(root, linkTarget)) {
+                return root
             }
         }
-        if (escapedRoot) {
-            return escapedRoot
+
+        return false
+    }
+
+    function fs_canEscape(
+        maybeLinkPath: string,
+        roots = defaultRoots
+    ): boolean {
+        // maybeLinkPath is the path which may be a symlink
+
+        if (!path.isAbsolute(maybeLinkPath)) {
+            maybeLinkPath = path.resolve(maybeLinkPath)
```

Comment:
@gregmagolan I think if we do this line 100% of the time then we don't have to worry about non-resolved/normalized paths being passed into the helpers, so the `path.sep` won't be an issue? For me that made my test go from a consistent 88s to 90s. But we might have to?

Reply:
Sounds like its all normalized. We just need to run CI on Windows to be sure.

Reply:
Moving the conversation to the other thread about the `/`s...